eth_utils/functional.py

```diff
@@ -2,23 +2,34 @@
 import functools
 import itertools
 
+from typing import (
+    Any,
+    Callable,
+)
+
 from .toolz import (
     compose as _compose,
 )
 
 
-def identity(value):
+AnyCallable = Callable[..., Any]
+
+
+def identity(value: Any) -> Any:
```

Comment:
I think we can make this more exact using generics.  Basically, this returns the type of whatever it is passed.

```python
T = typing.TypeVar('T')

def identity(value: T) -> T:
    return value
```